### ISSUE TITLE
How to use models defined in Tensorpack without using a Tensorpack trainer?

### ISSUE THREAD

**Issue by laoreja** [tensorpack/tensorpack#1476]
Hi there,

Thank you for the great repo and examples!

We just want to incorporate some model written in Tensorpack into some existing paper's official repo that did not use Tensorpack.

I'm able to extract the model definition and tried to run it with the training script of that repo, without using the Tensorpack trainers, since the other repo is really complicated itself.
The problem I met is that many layers and functions rely on the `get_current_tower_context()` function, e.g., BatchNorm and regularize_cost. And this makes the incorporation fail.

Are there any elegant solutions for this situation?

Thanks!

**Comment by ppwwyyxx** [comment]
see docs: https://tensorpack.readthedocs.io/tutorial/symbolic.html#use-models-outside-tensorpack
